runtime/src/bank.rs

```diff
@@ -6648,6 +6649,111 @@ impl Bank {
+            0xb9, 0xe1, 0x11, 0x5f,
+        ];
+        const VERIFIED_BUILD_HASH: Hash = Hash::new_from_array(HASH_BYTES);
+
+        let slashing_program_data_key =
+            bpf_loader_upgradeable::get_program_data_address(&SLASHING_PROGRAM_KEY);
+        let buffer_program_data_key =
+            bpf_loader_upgradeable::get_program_data_address(&BUFFER_PROGRAM_KEY);
+
+        if self
+            .get_account(&SLASHING_PROGRAM_KEY)
+            .map(|acc| !acc.data().is_empty())
+            .unwrap_or(false)
+        {
+            error!("Slashing program account is already populated");
+            return;
+        }
+
+        if self
+            .get_account(&slashing_program_data_key)
+            .map(|acc| !acc.data().is_empty())
+            .unwrap_or(false)
+        {
+            error!("Slashing program data account is already populated");
+            return;
+        }
+
+        let Some(buffer_program_account) = self.get_account(&BUFFER_PROGRAM_KEY) else {
+            error!("Buffer program account {} is missing", BUFFER_PROGRAM_KEY);
+            return;
+        };
+
+        let Some(buffer_program_data_account) = self.get_account(&buffer_program_data_key) else {
+            error!(
+                "Buffer program data account {} is missing",
+                buffer_program_data_key
+            );
+            return;
+        };
+
+        let (offset, end_offset, build_hash) =
+            Self::get_build_hash_and_offsets(buffer_program_data_account.data());
+        if build_hash != VERIFIED_BUILD_HASH {
+            error!(
+                "Buffer verified build hash {} does not match expected {}",
+                build_hash, VERIFIED_BUILD_HASH
+            );
+            return;
+        }
+
+        let mut slashing_program_account = buffer_program_account.clone();
+        slashing_program_account.set_owner(bpf_loader_upgradeable::id());
+        slashing_program_account
+            .set_state(&bpf_loader_upgradeable::UpgradeableLoaderState::Program {
+                programdata_address: slashing_program_data_key,
+            })
+            .unwrap();
+        slashing_program_account.set_executable(true);
+
+        let mut slashing_program_data_account = buffer_program_data_account.clone();
+        slashing_program_data_account
+            .set_state(
+                &bpf_loader_upgradeable::UpgradeableLoaderState::ProgramData {
+                    slot: self.slot,
+                    upgrade_authority_address: Some(system_program::id()),
+                },
+            )
+            .unwrap();
+        slashing_program_data_account.data_as_mut_slice()[offset..end_offset]
+            .copy_from_slice(&buffer_program_data_account.data()[offset..end_offset]);
+
+        self.store_account(&SLASHING_PROGRAM_KEY, &slashing_program_account);
+        self.store_account(&slashing_program_data_key, &slashing_program_data_account);
```

Comment:
In the past, we've typically zero'ed out the previous accounts, ie. `BUFFER_PROGRAM_KEY` and `buffer_program_data_key`, to avoid creating lamports. And we updated the capitalization accordingly.

Can you do that here too?

There's a few other things that we used to do at https://github.com/anza-xyz/agave/blob/eadab57a3a045a2a0d35857caec3d2dabcf90e7c/runtime/src/bank.rs#L6810-L6815, but I'm not sure if they're still needed